assembl/alembic/versions/7bf35c9261d9_delete_extra_fields_for_deleted_users.py

```diff
@@ -0,0 +1,30 @@
+"""delete_extra_fields_for_deleted_users
+
+Revision ID: 7bf35c9261d9
+Revises: e3f681bd70e0
+Create Date: 2018-06-13 14:14:41.037265
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '7bf35c9261d9'
+down_revision = 'e3f681bd70e0'
+
+from alembic import context, op
+import sqlalchemy as sa
+import transaction
+
+
+from assembl.lib import config
+
+
+def upgrade(pyramid_env):
+    from assembl import models as m
+    db = m.get_session_maker()()
+    with transaction.manager:
+        db.query(m.ProfileField).join(m.User).filter(m.User.is_deleted).delete()
```

Comment:
I would be explicit here `m.User.is_deleted == True`

Reply:
I'm not sure, maybe @vincentfretin can chime in, but confirm that this doesn't delete the joined table rows. If so, you can get the list of ProfileField id's, and delete each id.

Reply:
Putting `.delete()` at the end will generate a DELETE query like it should. If `.all()` returns ProfileField objects, then `.delete()` will remove exactly those objects.

Reply:
`m.User.is_deleted == True` is better yes.